pkgs/top-level/all-packages.nix

```
@@ -21553,6 +21553,14 @@ with pkgs;
 
   jxrlib = callPackage ../development/libraries/jxrlib { };
 
+  liana = callPackage ../by-name/li/liana {
+    withGui = true;
+  };
+
+  lianad = callPackage ../by-name/li/liana {
+    withGui = false;
+  };
+
```

nit: we could probably just oneline and group these by dropping the newline inbetween as
```suggestion
  liana = callPackage ../by-name/li/liana { withGui = true; };
  lianad = callPackage ../by-name/li/liana { withGui = false; };
```

---

Actually we only need `lianad` here as that is the virtual package:

`lianad = callPackage liana.override { withGui = false; };`

but the filename change above should be reverted.

---

You can see a similar setup for `lshw-gui` and `lshw` (just inverted bools)

---

looks like CI [doesn't like this approach](https://github.com/NixOS/nixpkgs/actions/runs/9753108716/job/26917825105?pr=323696#step:7:16):
```
- Attribute `pkgs.lianad` is a new top-level package using `pkgs.callPackage ... { /* ... */ }`.
  Please define it in pkgs/by-name/li/lianad/package.nix instead.
  See `pkgs/by-name/README.md` for more details.
  Since the second `callPackage` argument is not `{ }`, the manual `callPackage` in pkgs/top-level/all-packages.nix is still needed.

This PR introduces additional instances of discouraged patterns as listed above. Merging is discouraged but would not break the base branch.
```

---

Ah yes sorry, didn't delete the `callPackage` in my comment after copying the original.

Thanks!